spotfire/data_function.py

```diff
@@ -302,27 +302,24 @@ def _compile_script(self, result: AnalyticResult) -> None:
 
     def _read_inputs(self, result: AnalyticResult) -> None:
         """read inputs"""
-        self.debug("reading %d input variables" % len(self.inputs))
+        self.debug(f"reading {len(self.inputs)} input variables")
         for i, input_ in enumerate(self.inputs):
             if _bad_string(input_.name) or input_.name == "":
-                self.debug("input %d: bad input variable name - skipped" % i)
+                self.debug(f"input {int(i)}: bad input variable name - skipped")
```

Comment:
Probably don't need to convert an int to int before stringing it as part of the f-string.

Reply:
Good catches!  Yeah, the flynt package I used to do the mostly automatic conversions doesn't know when %d requires an explicit cast to int or not.  Thanks for catching these.  I'll adjust.